src/clarity-icons/extend-clarity-icons.ts

```diff
@@ -0,0 +1,98 @@
+import { IconTemplate } from "./icon-template";
+import { IconAlias } from "./icon-alias";
+import { SVG_IMG_ICON_TEMPLATES } from "./svg-img-icon-templates";
+import { SVG_ICON_TEMPLATES } from "./svg-icon-templates";
+
+let EXTENDED_ICON_TEMPLATES: IconTemplate = {};
+
+export class UserIcons {
+
+    private validateName(name: string): boolean {
+
+        return name.length !== 0;
+
+    }
+
+    private validateTemplate(template: string): boolean {
+
+        return template.length !== 0;
+
+    }
+
+    getExtendedShapes(): IconTemplate {
+
+        return EXTENDED_ICON_TEMPLATES;
+    }
+
+
+    private setIconTemplate(shapeName: string, shapeTemplate: string): void {
+
+        if (this.validateName(shapeName) && this.validateTemplate(shapeTemplate)) {
+
+            EXTENDED_ICON_TEMPLATES[ shapeName ] = shapeTemplate;
+        }
+
+    }
+
+    addIcons(icons: IconTemplate): void {
+
+        for (let shapeName in icons) {
+
+            if (icons.hasOwnProperty(shapeName)) {
+
+                this.setIconTemplate(shapeName, icons[ shapeName ]);
+            }
+        }
+
+    }
+
+    private setIconAliases(templates: IconTemplate, shapeName: string, aliasNames: string[]): void {
+        for (let aliasName of aliasNames) {
+
+            Object.defineProperty(templates, aliasName, {
+                get: function () {
+                    return templates[ shapeName ];
+                },
+                enumerable: true,
+                configurable: true
+            });
+        }
+    }
+
+    addAliases(aliases: IconAlias): void {
```

Comment:
Sure. Btw, this is how `ClarityIcons.addAliases` works:

```
ClarityIcons.addAliases({caret: ["arrow", "chevron", "my-caret"]});
```

So `caret` will have aliases such as `arrow`, `chevron`, and `my-caret`.